commit 7fbe8a36047728cc1204fb1af68f9ed900affada
Author: Di Wang <ddiwang@google.com>
Date:   Mon Feb 10 18:08:51 2025 +0000

    prov/tcp: Only disable ep if the failure can not be retried
    
    Only disable ep if the failure can not be retried.
    
    Signed-off-by: Di Wang <ddiwang@google.com>

diff --git a/src/common.c b/src/common.c
index 0d641ac74..b190e52e0 100644
--- a/src/common.c
+++ b/src/common.c
@@ -1421,6 +1421,9 @@ int ofi_bsock_async_done(const struct fi_provider *prov,
 	msg.msg_controllen = sizeof(ctrl);
 	ret = recvmsg(bsock->sock, &msg, MSG_ERRQUEUE);
 	if (ret < 0) {
+		if (OFI_SOCK_TRY_SND_RCV_AGAIN(errno))
+			return 0;
+
 		FI_WARN(prov, FI_LOG_EP_DATA,
 			"Error reading MSG_ERRQUEUE (%s)\n", strerror(errno));
 		return -errno;
